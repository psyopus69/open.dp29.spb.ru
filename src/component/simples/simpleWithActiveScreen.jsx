import {PropTypes} from "prop-types";
import React from "react";

const SimpleWithActiveScreen = (props) => {
  const {renderScreen, changeScreen} = props;
  return <React.Fragment>
    <div>Change screen to id = -1 : <button onClick={() => {
      changeScreen(-1);
    }}> to -1 </button></div>
    <div>Change screen ++ : <button onClick={() => {
      changeScreen();
    }}>screen ++</button></div>
    <div>Screen :</div>{renderScreen()}
  </React.Fragment>;
};
SimpleWithActiveScreen.propTypes = {
  renderScreen: PropTypes.func.isRequired,
  changeScreen: PropTypes.func.isRequired,
};

export {
  SimpleWithActiveScreen,
};
