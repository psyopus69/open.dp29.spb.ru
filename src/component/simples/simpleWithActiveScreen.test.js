import React from "react";
import renderer from "react-test-renderer";
import {SimpleWithActiveScreen} from "./simpleWithActiveScreen.jsx";

it(`with methods  renderScreen() & changeScreen(id | ++)`, () => {
  const tree = renderer.create(<SimpleWithActiveScreen
    renderScreen={jest.fn()}
    changeScreen={jest.fn()}
  />);
  expect(tree).toMatchSnapshot();
});
