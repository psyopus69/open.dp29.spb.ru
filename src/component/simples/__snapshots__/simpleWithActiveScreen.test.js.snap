// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with methods  renderScreen() & changeScreen(id | ++) 1`] = `
Array [
  <div>
    Change screen to id = -1 : 
    <button
      onClick={[Function]}
    >
       to -1 
    </button>
  </div>,
  <div>
    Change screen ++ : 
    <button
      onClick={[Function]}
    >
      screen ++
    </button>
  </div>,
  <div>
    Screen :
  </div>,
]
`;
